lms/djangoapps/branding/api.py

```diff
@@ -194,111 +197,135 @@ def _build_support_form_url():
     return '{base_url}/support/contact_us'.format(base_url=settings.LMS_ROOT_URL)
 
 
-def _footer_navigation_links():
+def _find_position_of_link(links, key):
+    "Returns position of the link to be inserted"
+    for link in links:
+        if link[0] == key:
+            return links.index(link) + 1
+
+
+def _footer_navigation_links(language=settings.LANGUAGE_CODE):
     """Return the navigation links to display in the footer. """
     platform_name = configuration_helpers.get_value('platform_name', settings.PLATFORM_NAME)
+    links = [
+        ("about", (marketing_link("ABOUT"), _("About"))),
+        ("enterprise", (marketing_link("ENTERPRISE"),
+                        _("{platform_name} for Business").format(platform_name=platform_name))),
+        ("blog", (marketing_link("BLOG"), _("Blog"))),
+        ("help-center", (settings.SUPPORT_SITE_LINK, _("Help Center"))),
+        ("contact", (reverse("support:contact_us"), _("Contact"))),
+        ("careers", (marketing_link("CAREERS"), _("Careers"))),
+        ("donate", (marketing_link("DONATE"), _("Donate"))),
+    ]
+
+    if language == settings.LANGUAGE_CODE:
+        position = _find_position_of_link(links, 'blog')
```

Comment:
I belive you already know the position of `blog` just use that instead of finding it through code.